requirements-min.txt

```diff
@@ -1,6 +1,6 @@
 # package dependencies and their minimum versions for installing HDMF
 # the requirements here specify '==' for testing; setup.py replaces '==' with '>='
-h5py==2.9,<3  # support for setting attrs to lists of utf-8 added in 2.9
+h5py==3.1.0  # support for setting attrs to lists of utf-8 added in 2.9
```

Comment:
Do we need to cut h5py 2.9-2.10 compatibility already? I think we should try to maintain compatibility for some time, at least until ubuntu supports h5py 3.x. 

Ubuntu: current support 2.10: https://packages.ubuntu.com/search?suite=groovy&searchon=names&keywords=h5py
Fedora currently supports 3.1: https://src.fedoraproject.org/rpms/h5py
Debian currently supports 2.8: https://packages.debian.org/search?keywords=h5py&searchon=names&suite=stable&section=all

Reply:
I'm not opposed to it, but we should discuss how we decide when to stop supporting 2.x.

Reply:
Ideally (unless too much effort) it would be great to retain compatibility with h5py 2.x since even if bleeding edge releases of distributions would support it, stable releases would not